src/gov/nist/javax/sip/parser/ims/PAccessNetworkInfoParser.java

```diff
@@ -105,43 +106,53 @@ public SIPHeader parse() throws ParseException
+                accessNetworkInfo.setHeaderName(SIPHeaderNamesIms.P_ACCESS_NETWORK_INFO);
 
-            this.lexer.SPorHT();
-            lexer.match(TokenTypes.ID);
-            Token token = lexer.getNextToken();
-            accessNetworkInfo.setAccessType(token.getTokenValue());
-
-            this.lexer.SPorHT();
-            while (lexer.lookAhead(0) == ';') {
-                this.lexer.match(';');
                 this.lexer.SPorHT();
-
-                try {
-                	NameValue nv = super.nameValue('=');
-                	accessNetworkInfo.setParameter(nv);
-                } catch (ParseException e) {
-                	this.lexer.SPorHT();
-                	String ext = this.lexer.quotedString();
-                	if(ext == null) {
-                		ext = this.lexer.ttokenGenValue();
-                	} else {
-                		// avoids tokens such as "a=b" to be stripped of quotes and misinterpretend as
-                		// RFC 7913 generic-param when re-encoded
-                		ext = "\"" + ext + "\""; 
-                	}
-                	accessNetworkInfo.setExtensionAccessInfo(ext);
+                lexer.match(TokenTypes.ID);
+                Token token = lexer.getNextToken();
+                accessNetworkInfo.setAccessType(token.getTokenValue());
+                this.lexer.SPorHT();
+                while (lexer.lookAhead(0) == ';') {
+                    this.lexer.match(';');
+                    this.lexer.SPorHT();
+
+                    try {
+                        NameValue nv = super.nameValue('=');
+                        accessNetworkInfo.setParameter(nv);
+                    } catch (ParseException e) {
+                        this.lexer.SPorHT();
+                        String ext = this.lexer.quotedString();
+                        if (ext == null) {
+                            ext = this.lexer.ttokenGenValue();
+                        } else {
+                            // avoids tokens such as "a=b" to be stripped of quotes and misinterpretend as
+                            // RFC 7913 generic-param when re-encoded
+                            ext = "\"" + ext + "\"";
+                        }
+                        accessNetworkInfo.setExtensionAccessInfo(ext);
+                    }
+                    this.lexer.SPorHT();
                 }
+                accessNetworkInfoList.add(accessNetworkInfo);
                 this.lexer.SPorHT();
+                char la = lexer.lookAhead(0);
+                if (la == ',') {
+                    this.lexer.match(',');
+                    this.lexer.SPorHT();
+                } else if (la == '\n')
+                    break;
+                else
+                    throw createParseException("unexpected char");
             }
-            this.lexer.SPorHT();
-            this.lexer.match('\n');
-
-
-            return accessNetworkInfo;
+            return accessNetworkInfoList;
```

Comment:
is this going to break existing clients that are using the previous simple PAccessNetworkInfo class?

Reply:
I integrated with sip-servlet and test for both single and mutilple PANIs, it's still working.